pyomo/core/base/external.py

```diff
@@ -111,6 +111,13 @@ def __init__(self, *args, **kwds):
         self._so = None
         ExternalFunction.__init__(self, *args, **kwds)
 
+    def __getstate__(self):
+        state = super(AMPLExternalFunction, self).__getstate__()
+        # Remove reference to loaded library (they are not copyable or
+        # picklable)
+        state['_so'] = state['_known_functions'] = None
+        return state
+
```

Comment:
I assume that removing the reference to the loaded library ensures it will be reloaded when required?

Reply:
That is correct.  Both `_so` and `_known_functions` are initialized to `None` in `__init__()`, and `_evaluate()` begins with:
```python
if self._so is None:
    self.load_library()
```